pkg/cmd/cmd_add.go

```diff
@@ -99,17 +85,93 @@ func onlyOnceOption(c *cli.Context, name string) (*string, error) {
 	} else if len(inputOpt) == 1 {
 		return &inputOpt[0], nil
 	} else {
-		reporter.Report("kpm: the following required arguments were not provided: ", name)
-		reporter.ExitWithReport("kpm: run 'kpm add help' for more information.")
-		return nil, fmt.Errorf("kpm: Invalid command")
+		return nil, nil
 	}
 }
 
-func addGitDep(opt *opt.AddOptions, kclPkg *pkg.KclPkg) error {
-	if opt.RegistryOpts.Git == nil {
-		reporter.Report("kpm: a value is required for '-git <URI>' but none was supplied")
-		reporter.ExitWithReport("kpm: run 'kpm add help' for more information.")
+// parseAddOptions will parse the user cli inputs.
+func parseAddOptions(c *cli.Context, localPath string) (*opt.AddOptions, error) {
+	// parse from 'kpm add -git https://xxx/xxx.git -tag v0.0.1'.
+	if c.NArg() == 0 {
+		gitOpts, err := parseGitRegistryOptions(c)
+		if err != nil {
+			return nil, err
+		}
+		return &opt.AddOptions{
+			LocalPath:    localPath,
+			RegistryOpts: *gitOpts,
+		}, nil
+	} else {
+		// parse from 'kpm add xxx:0.0.1'.
+		ociReg, err := parseOciRegistryOptions(c)
+		if err != nil {
+			return nil, err
+		}
+		return &opt.AddOptions{
+			LocalPath:    localPath,
+			RegistryOpts: *ociReg,
+		}, nil
+	}
+
+}
+
+// parseGitRegistryOptions will parse the git registry information from user cli inputs.
+func parseGitRegistryOptions(c *cli.Context) (*opt.RegistryOptions, error) {
+	gitUrl, err := onlyOnceOption(c, "git")
```

Comment:
The feeling here is inconsistent with the OCI's options parse process. Ref:  https://buf.build/docs/tutorials/getting-started-with-bsr/

Reply:
> The feeling here is inconsistent with the OCI's options parse process. Ref: https://buf.build/docs/tutorials/getting-started-with-bsr/

Do we necessarily need git and tag options?

Reply:
This implementation references the cargo command, which can be adjusted later to separate the git related content from the OCI registry into two different commands. Or if adding a git repository as a dependency is not needed, git related content can be removed later.